packages/loader/container-loader/api-report/container-loader.api.md

```diff
@@ -126,9 +181,31 @@ export class Loader implements IHostLoader {
     readonly services: ILoaderServices;
 }
 
+// @beta (undocumented)
+export interface PendingConnectionStep {
+    // (undocumented)
+    name: string;
+    // (undocumented)
+    retryableError?: IAnyDriverError;
+    // (undocumented)
+    time: number;
+    // (undocumented)
+    type?: "auth" | "socket.io" | "orderingService";
```

Comment:
Do we expect all steps to fall in one of these buckets? And are these 3 an exhaustive list of the buckets or is this just a starting list that will be updated during implementation?

Reply:
This definitely needs refinement, and input from partners like you. The idea would be to identify a constrained number of buckets here that have semantic meaning that can be coded against by a partner team, e.g. to trigger different UX.

It's possible we would even start with nothing, and only after spending time working with the raw names and discussing end-to-end scenarios would we start to build this set.